### PR TITLE
Task#5257 - Follow user

### PR DESCRIPTION
**Pull request by taylq**
https://edu-redmine.framgia.vn/issues/5257?issue_count=28&issue_position=4&next_issue_id=5256&prev_issue_id=5258

### CODE REVIEW

**Comment by taylq** [comment]
ready

File: app/models/relationship.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (huongnguyenmta) [inline app/models/relationship.rb]: => `class_name: User.name`

File: app/models/user.rb [hunk 2]
```diff
@@ -4,6 +4,12 @@ class User < ApplicationRecord
 
   has_many :borrows, dependent: :destroy
   has_many :comments, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
```
Comment (huongnguyenmta) [inline app/models/user.rb]: tương tự

File: app/views/relationships/destroy.js.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$("#follow_form").html("<%= escape_javascript(render("users/follow")) %>");
```
Comment (huongnguyenmta) [inline app/views/relationships/destroy.js.erb]: => `j render ...`

File: app/views/users/_follow.html.erb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<%= form_for current_user.active_relationships.build, remote: true do |f| %>
```
Comment (huongnguyenmta) [inline app/views/users/_follow.html.erb]: `current_user.active_relationships.build` xử lý trong controller em nhé

File: app/views/shared/_stats.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<% @user ||= current_user %>
```
Comment (huongnguyenmta) [inline app/views/shared/_stats.html.erb]: sao gán ở đây vậy em?

File: app/views/users/_unfollow.html.erb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<%= form_for current_user.active_relationships.find_by(followed_id: @user.id),
```
Comment (huongnguyenmta) [inline app/views/users/_unfollow.html.erb]: ` current_user.active_relationships.find_by(followed_id: @user.id)`: xử lý trong controller

**Comment by taylq** [comment]
ready

**Comment by taylq** [comment]
ready

**Pull request merged**